src/grids/Grid.jl

```diff
@@ -31,13 +31,13 @@ function make_cartesian_grid(problem, size::Int64)
     xmin, xmax = domain
     nx = size
     dx1 = (xmax - xmin) / nx
-    xc = LinRange(xmin + 0.5 * dx1, xmax - 0.5 * dx1, nx)
+    xc = collect(LinRange(xmin + 0.5 * dx1, xmax - 0.5 * dx1, nx))
```

Comment:
Is this 0.5 here okay? Will this ruin the type, e.g., when Float32 is used? Check everywhere.